core/trace/src/main/scala/org/typelevel/otel4s/trace/Tracer.scala

```diff
@@ -172,6 +174,8 @@ trait Tracer[F[_]] extends TracerMacro[F] {
     */
   def noopScope[A](fa: F[A]): F[A]
 
+  def mapK[G[_]: Sync](fk: F ~> G, gk: G ~> F): Tracer[G]
```

Comment:
Here the type constraint must satisfy `SpanRunner` and `TraceScope` from the java-trace package.